slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -106,10 +106,26 @@ void SLAMServiceImpl::SetUpMapBuilder() {
     mapBuilder.BuildMapBuilder();
 }
 
-void SLAMServiceImpl::CreateMap() {
+void SLAMServiceImpl::ProcessDataOffline() {
     // Set up and build the MapBuilder
     SetUpMapBuilder();
 
+    // TODO: Check whether or not we're running localization and/or
+    // updating a map. Create a function to run those modes.
+    // Implemented with these tickets:
+    // https://viam.atlassian.net/browse/DATA-114
+    // https://viam.atlassian.net/browse/DATA-631
+
+    // TODO: Call localization only mode function, if localization only
+    // action mode
+
+    // TODO: Call updating map function, if apriori map was loaded
+
+    // Mapping a new map:
+    CreateMap();
+}
+
+void SLAMServiceImpl::CreateMap() {
```

Comment:
Would it make sense to delete this function now, in favor of `ProcessDataOffline()`?

Reply:
I don't think so. `CreateMap` handles creating a map from scratch. When there is an apriori map, or when we're localizing only, the function looks differently. That's why I wrapped `CreateMap` within `ProcessDataOffline`, and we'll add if/else statements later on that'll execute a different function instead when updating/localizing.

Reply:
sounds good!